src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,108 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 0;
+    private static final int DEFAULT_SIZE = 0;
+    private Object[] array;
+    private int size;
+
+    public ArrayList() {
+        array = new Object[DEFAULT_CAPACITY];
+        size = DEFAULT_SIZE;
```

Comment:
It's up to you but I would rather start with larger `DEFAULT_CAPACITY` so we will have less iterations of growing the initial array.
Also, initializing size as `private static final int DEFAULT_SIZE = 0;` is redundant. `int` by default is 0.

Reply:
1. Initial capacity: 
This can be beneficial for performance, especially in situations where the list is expected to grow significantly. 
2. Initializing:
I prefer explicitly initializing all fields to make the code more explicit and self-documenting.